drivers/char/broadcom/vc_sm/vmcs_sm.c

```diff
@@ -1256,61 +1256,106 @@ static const struct vm_operations_struct vcsm_vm_ops = {
 	.fault = vcsm_vma_fault,
 };
 
-/* Walks a VMA and clean each valid page from the cache */
-static void vcsm_vma_cache_clean_page_range(unsigned long addr,
-					    unsigned long end)
+static int clean_invalid_mem_2d(const void __user *addr,
+		const size_t block_count, const size_t block_size, const size_t stride,
+		const unsigned cache_op)
 {
-	pgd_t *pgd;
-	pud_t *pud;
-	pmd_t *pmd;
-	pte_t *pte;
-	unsigned long pgd_next, pud_next, pmd_next;
+	size_t i;
+	void (*op_fn)(const void*, const void*);
 
-	if (addr >= end)
-		return;
+	if (block_size <= 0) {
```

Comment:
I was writing the condition so that it can detect negative size if the size was unintentionally signed, but it's obviously unsigned here, so fixed in https://github.com/raspberrypi/linux/pull/2333/commits/872f3ca2c4dd268bc6d537151435e45f16224b0e.